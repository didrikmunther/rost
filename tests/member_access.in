// This program is the test program

struct MyOtherStruct {
	k: int,
}

struct MyOtherStruct2 {
	k: int,
}

struct MyStruct {
	a: int,
	b: int,
	c: int,
	o: &MyOtherStruct,
	o2: MyOtherStruct
}

fn main(argc: int, argv: &&char) {

	let o = MyOtherStruct {
		k: 123
	};

	let s: MyStruct = MyStruct {
		a: 1,
		b: 2,
		c: 3,
		o: &o,
		o2: MyOtherStruct {
			k: 321
		}
	};

	s.a = (*s.o).k + 10;
	(*s.o).k = 666;
	s.o2.k = s.a + 5;

	printf("%i\n", s.a);
	printf("%i\n", s.b);
	printf("%i\n", s.c);
	printf("%i\n", (*s.o).k);
	printf("%i\n", s.o2.k);
}