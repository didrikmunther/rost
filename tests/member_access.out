133
2
3
668
668
138